scripts/ci/make/util

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env bash
+
+#########################################################
+###   COMMON UTILITY FUNCTIONS - ALPHABETICAL ORDER   ###
+#########################################################
+
+cleanup() {
+    rm -rf "${VENV}/tmp"
+}
+
+clean_container() {
+    print_colorized ERROR "deleting ${TEST_IMAGE_NAME}"
+    sleep 3
+    docker rmi "${TEST_IMAGE_NAME}"
+}
+
+clean_python_env() {
+    print_colorized ERROR "deleting virtualenv & build data"
+    sleep 3
+    rm -rf "${VENV}" \
+        *.egg-info \
+        dist \
+        build
+    find . -name '*.pyc' -exec rm -f {} \;
+}
+
+clean_tests() {
+    print_colorized ERROR "deleting testing temp data"
+    sleep 3
+    rm -rf .tox
+    rm -f tox.log
+    rm -rf .pytest_cache
+}
+
+docker_compose_down() {
+    docker-compose -f scripts/ci/docker-compose-ci.yaml down
+}
+
+docker_compose_up() {
+    docker-compose -f scripts/ci/docker-compose-ci.yaml up -d
+}
+
+install() {
+    print_colorized WARN "installing ${GIT_REPO} to ${VENV}"
+    pip install .
+}
+
+install_dev() {
+    print_colorized WARN "installing ${GIT_REPO} in editable mode to ${VENV}"
+    pip install --editable .
+}
+
+install_cluster_deps() {
+    local arch=$(uname | tr A-Z a-z)
+
+    if [[ ! -x "${VENV}/bin/kind" ]]; then
+        print_colorized WARN "installing kind"
+        curl -qsSLo "${VENV}/bin/kind" "https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-${arch}-amd64"
+        chmod +x "${VENV}/bin/kind"
+    fi
+    if [[ ! -x "${VENV}/bin/helm" ]]; then
+        print_colorized WARN "installing helm"
+        curl -sSL "https://get.helm.sh/helm-${HELM_VERSION}-${arch}-amd64.tar.gz" | tar xzf - -C "${VENV}/bin" --strip-components=1 "${arch}-amd64/helm"
+        chmod +x "${VENV}/bin/helm"
+    fi
+    if [[ ! -x "${VENV}/bin/kubectl" ]]; then
+        print_colorized WARN "installing kubectl"
+        curl -sSLo "${VENV}/bin/kubectl" "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/${arch}/amd64/kubectl"
+        chmod +x "${VENV}/bin/kubectl"
+    fi
+}
+
+install_testing_deps() {
+    print_colorized WARN "installing dependencies: $*"
+    mkdir -p .tox
+    for p in "$@"; do
+        if [[ ! "$(echo ${VENV}/bin/*)" =~ "$p" ]]; then
```

Comment:
is this trying to match the dependency to the script name? If so... there are many situations where this doesn't hold true

Reply:
It's trying to find if any of the binaries in your venv match the name of the dependency. I was trying to have it not install the dependency if it already exists in your venv. In particular, the issue was that anchore-cli is installed using the pip package `anchorecli` but the binary is called `anchore-cli` so I couldn't do `hash $p` to see if it was availabe in your PATH. This is one of the few situations where ansible would be really helpful. Or maybe you know another bash trick to make this determination?

Reply:
If you really wanted to create a faithful delta on what is installed vs. what is being asked to install, `pip freeze` could get the output and then a `grep` to see if it is there. Alternatively, this could just do without the trouble of doing the check and install. `pip` will not re-install if the package is already there.

And yes, a framework like Ansible would prevent this situation with:

```yaml
- pip:
    name: anchorecli
    virtualenv: { VENV }
```